azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -152,25 +187,29 @@ def run(self, goal_state, ext_gs_updated):
 
             agent_family = self._get_agent_family_manifest(goal_state)
 
-            # Updater will return True or False if we need to switch the updater
-            # If self-updater receives RSM update enabled, it will switch to RSM updater
-            # If RSM updater receives RSM update disabled, it will switch to self-update
-            # No change in updater if GS not updated
-            is_rsm_update_enabled = self._updater.is_rsm_update_enabled(agent_family, ext_gs_updated)
+            # Always agent uses self-update for initial update regardless vm enrolled into RSM or not
+            # So ignoring the check for updater switch for the initial goal state
+            if self._is_initial_update_attempted():
```

Comment:
can we rename this method (and change its return value)? the current name makes this code very easy to misread. maybe invert its return value and name it "_is_initial_update"? then this code would read as "if not self._is_initial_update():"